ls1intum/Artemis
# [Exam] Add descriptive exam statistics

### Checklist

### Motivation and Context
This PR adds more exam statistics to give the instructor a better overview of the results. The statistics will be visualized with diagrams in a follow up.

### Description

- [x] Add descriptive exam statistics
- [x] Add score distribution, which will be used to create a histogram
- [x] Add dependency for simple statistics https://bundlephobia.com/result?p=simple-statistics@7.1.0. This can be used to calculate some statistic measurements. Small enough with big community (>50k weekly downloads)

### Steps for Testing


1. Create an exam with all exercise types, and multiple exercises per exercise group. You can also tryout optional exercise groups.
2. Conduct the exam with multiple users. Do not work on some exercise and do not submit the exam for some users. Note down what you did with which user.
3. Assess the exam
4. Check the exam scores page and play around with filters
--> Check that the statistics are correctly calculated

### Screenshots


![Capture](https://user-images.githubusercontent.com/16407766/89834235-b64d0400-db62-11ea-90b4-b644b2ff2fc3.PNG)
